pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -2084,18 +2084,20 @@ public JSONObject rebalanceTable(final String rawTableName, TableType tableType,
     try {
       RebalanceSegmentStrategy rebalanceSegmentsStrategy =
           RebalanceSegmentStrategyFactory.getInstance().getRebalanceSegmentsStrategy(tableConfig);
-      PartitionAssignment newPartitionAssignment =
-          rebalanceSegmentsStrategy.rebalancePartitionAssignment(idealState, tableConfig, rebalanceUserConfig);
-      IdealState targetState =
-          rebalanceSegmentsStrategy.getRebalancedIdealState(idealState, tableConfig, rebalanceUserConfig);
-
-      // update it as needed
+      // rebalance it as needed
       if (!rebalanceUserConfig.getBoolean(RebalanceUserConfigConstants.DRYRUN)) {
         // call the updater providing it a reference to the strategy so it can recompute target state if needed
-        _idealStateUpdater.update(targetState, tableConfig, rebalanceSegmentsStrategy, rebalanceUserConfig);
+        _tableRebalancer.rebalance(tableConfig, rebalanceSegmentsStrategy, rebalanceUserConfig);
+        jsonObject.put("status", "Rebalance triggered. Check logs for progress");
+      } else {
+        // dry run mode: return the target partition assignment and idealstate mapping
```

Comment:
can we do this part of tableRebalancer itself?  rebalance method should return the final idealstate.